### PR TITLE
feat: Improved various bundles

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

* Refactor entrypoints so that they are clearer and more automatable
* Sadly couldn't just user the file extensions `.mts` as tsc doesn't like it so used a sub extension to indicate target
* Add more stuff to the `full` file
* Adds a config option for explicitly stating that you never want to load external scripts

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: src/entrypoints/exception-autocapture.ts
Comment (pauldambra) [inline src/entrypoints/exception-autocapture.ts:1]: want me to update this? it's changed in recent PRs

File: src/entrypoints/array.full.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Same as loader-globals.ts except includes all additional extension loaders
+
+import './recorder'
+import './surveys'
+import './exception-autocapture'
+import './tracing-headers'
```
Comment (pauldambra) [inline src/entrypoints/array.full.ts:6]: and web vitals too now

File: package.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
         "prepare": "husky install",
         "deprecate-old-versions": "node scripts/deprecate-old-versions.mjs"
     },
-    "main": "dist/module.js",
-    "module": "dist/es.js",
+    "main": "dist/main.js",
+    "module": "dist/module.js",
```
Comment (pauldambra) [inline package.json:27]: ELI5? 

(i have managed to ignore changes to JS bundling over the last few chaotic years 🤣)

**Comment by pauldambra** [comment]
feels like there are some QoL improvements you could pull out of this (like the loadScript refactor) that would make it easier to ship them and then easier to review this 🙈

File: src/entrypoints/exception-autocapture.ts
Comment (pauldambra) [reply src/entrypoints/exception-autocapture.ts:1]: updated two bits - broke tests - i'll follow up when i've figured out why 🙈

Comment (pauldambra) [reply src/entrypoints/exception-autocapture.ts:1]: sorted

File: package.json (hunk 2 above)
Comment (benjackwhite) [reply package.json:27]: I think we just had it poorly named beforehand. The module wasn't really the module loader 🤷

**Comment by benjackwhite** [comment]
> feels like there are some QoL improvements you could pull out of this (like the loadScript refactor) that would make it easier to ship them and then easier to review this 🙈

That's fair. Didn't want to spend too much time here... But yeah I can split out the bundle changes from the functional changes

**Review by pauldambra — APPROVED** [review]
runs locally

**Comment by pauldambra** [comment]
Screenshot 2024-06-26 at 15 31 49

this removal ok?

**Comment by pauldambra** [comment]
i suppose only risk here is someone is importing from `dist/es.js` and their build breaks when updating?

**Comment by benjackwhite** [comment]
> i suppose only risk here is someone is importing from `dist/es.js` and their build breaks when updating?

I think thats fine, especially as it is npm only in that case so they would have the minor upgrade and we don't document that as an option anywhere
